app/src/main/java/org/eyeseetea/malariacare/layout/utils/LayoutUtils.java

```diff
@@ -219,4 +261,89 @@ public static String getAppNameColorString() {
+
+        final DoublePieChart doublePieChart =
+                (DoublePieChart) actionBar.getCustomView().findViewById(R.id.action_bar_chart);
+        doublePieChart.setVisibility(View.VISIBLE);
+        ISurveyAnsweredRatioRepository surveyAnsweredRatioRepository =
+                new SurveyAnsweredRatioRepository();
+        SaveSurveyAnsweredRatioUseCase saveSurveyAnsweredRatioUseCase =
+                new SaveSurveyAnsweredRatioUseCase(surveyAnsweredRatioRepository);
+        saveSurveyAnsweredRatioUseCase.execute(surveyId,
+                new ISurveyAnsweredRatioCallback() {
+                    @Override
+                    public void nextProgressMessage() {
+                        Log.d(getClass().getName(), "nextProgressMessage");
+                    }
+
+                    @Override
+                    public void onComplete(SurveyAnsweredRatio surveyAnsweredRatio) {
+                        Log.d(getClass().getName(), "onComplete");
+
+                        if (surveyAnsweredRatio != null) {
+                            updateChart(surveyAnsweredRatio, doublePieChart);
+                        }
+                    }
+                });
+    }
+
+    public static void updateActionBarChartCompletionCount(ActionBar actionBar, Long surveyId,
+            final boolean isCompulsory, final boolean visible) {
+        final DoublePieChart doublePieChart =
+                (DoublePieChart) actionBar.getCustomView().findViewById(R.id.action_bar_chart);
+        doublePieChart.setVisibility(View.VISIBLE);
+        final ISurveyAnsweredRatioRepository surveyAnsweredRatioRepository =
+                new SurveyAnsweredRatioRepository();
+        final GetSurveyAnsweredRatioUseCase getSurveyAnsweredRatioUseCase =
+                new GetSurveyAnsweredRatioUseCase(surveyAnsweredRatioRepository);
+        getSurveyAnsweredRatioUseCase.execute(surveyId,
+                new ISurveyAnsweredRatioCallback() {
+                    @Override
+                    public void nextProgressMessage() {
+                        Log.d(getClass().getName(), "nextProgressMessage");
+                    }
+
+                    @Override
+                    public void onComplete(SurveyAnsweredRatio surveyAnsweredRatio) {
+                        Log.d(getClass().getName(), "onComplete");
+                        surveyAnsweredRatio.fixTotalQuestion(isCompulsory, visible);
+                        saveAndShowPie(surveyAnsweredRatio, surveyAnsweredRatioRepository,
+                                doublePieChart);
+                    }
+                });
+    }
+
+    public static void saveAndShowPie(SurveyAnsweredRatio surveyAnsweredRatio,
+            ISurveyAnsweredRatioRepository surveyAnsweredRatioRepository,
+            final DoublePieChart doublePieChart) {
+        SaveSurveyAnsweredRatioUseCase saveSurveyAnsweredRatioUseCase = new SaveSurveyAnsweredRatioUseCase(surveyAnsweredRatioRepository);
```

Comment:
saveSurveyAnsweredRatioUseCase should not be executed from layoutUtils because is not layout or utils